plugin/ui/completion.py

```diff
@@ -323,45 +324,50 @@ class _PhantomCompletion(_BaseCompletion):
     def __init__(self, view: sublime.View) -> None:
         super().__init__(view)
 
-        self._phantom_set = self._get_phantom_set(view, True)
+        self._phantom_set = self._get_phantom_set(view)
 
     @classmethod
-    def _get_phantom_set(cls, view: sublime.View, init: bool = False) -> Optional[sublime.PhantomSet]:
+    def _get_phantom_set(cls, view: sublime.View) -> sublime.PhantomSet:
         view_id = view.id()
-        phantom_set = _phantom_sets_per_view.get(view_id, None)
 
-        if init:
-            phantom_set = sublime.PhantomSet(view, cls.COPILOT_PHANTOM_COMPLETION)
-            _phantom_sets_per_view[view_id] = phantom_set
+        # create phantom set if there is no existing one
+        if not _view_to_phantom_set.get(view_id):
+            _view_to_phantom_set[view_id] = sublime.PhantomSet(view, cls.COPILOT_PHANTOM_COMPLETION)
 
-        return phantom_set
+        return _view_to_phantom_set[view_id]
 
     def normalize_phantom_line(self, line: str) -> str:
-        # return re.sub(r"( {2,})", lambda m: "&nbsp;" * len(m.group(0)), html.escape(line))
-        return (
-            html.escape(line).replace(" ", "&nbsp;").replace("\t", "&nbsp;" * self.view.settings().get("tab_size", 4))
-        )
+        return html.escape(line).replace(" ", "&nbsp;").replace("\t", "&nbsp;" * self.view.settings().get("tab_size"))
 
     def _build_phantom(
-        self, lines: Union[str, List[str]], begin: int, end: Optional[int] = None, inline: bool = True
+        self,
+        lines: Union[str, Sequence[str]],
+        begin: int,
+        end: Optional[int] = None,
+        *,
+        inline: bool = True
+        # format separator
     ) -> sublime.Phantom:
+        view_settings = self.view.settings()
```

Comment:
Just thinking out loud, is it worth adding `self._settings = self.view.settings()` in the `__init__` ?

We also use view settings in the `normalize_phantom_line` method

Reply:
done.